frontend/src/layout/header/Search.tsx

```diff
@@ -62,13 +63,23 @@ export const SearchField: React.FC<SearchFieldProps> = ({ variant }) => {
     useEffect(() => () => clearTimeout(lastTimeout.current));
 
     const onSearchRoute = isSearchActive();
-    const defaultValue = onSearchRoute
-        ? new URLSearchParams(document.location.search).get("q") ?? undefined
-        : undefined;
-
-    const search = (expression: string) => {
-        router.goto(SearchRoute.url({ query: expression }), onSearchRoute);
+    const getSearchParam = (searchParameter: string) => {
+        const searchParams = new URLSearchParams(document.location.search);
+        return onSearchRoute
+            ? searchParams.get(searchParameter) ?? undefined
+            : undefined;
     };
+    const defaultValue = getSearchParam("q");
+
+
+    const search = useCallback(debounce((expression: string) => {
+        const filters = {
+            itemType: getSearchParam("f") as Item,
```

Comment:
This should use `validType` as well right? Generally, `as` casts should be seen as dangerous as it's basically "TypeScript, trust me". They can certainly be used but whenever writing one, one should make sure that it's actually valid.

Also, I would probably rename `validType` to `isValidItemType` or even `isValidSearchItemType` given that it's exported.